griptape/utils/stream.py

```diff
@@ -64,8 +64,11 @@ def event_handler(event: BaseEvent) -> None:
             handler=event_handler,
             event_types=[CompletionChunkEvent, FinishPromptEvent, FinishStructureRunEvent],
         )
+
         self.structure.add_event_listener(stream_event_listener)
+        self.structure.config.prompt_driver.add_event_listener(stream_event_listener)
```

Comment:
Will this work for all cases?

Reply:
Can you expand on which other cases you're envisioning?

Reply:
I suppose I was thinking about if prompt_driver is passed directly to structure, but I see now that that will implicitly create a config with the prompt driver, so I guess its not an issue.

Sorry for the vague comment, but I get the feeling like there are some edge cases that I'm missing. I suppose there probably aren't any for this particular usage since we know exactly which entities publish the event types listed above (just structure and prompt driver)